### DISCUSSION TITLE
Is there an option to fix a weird behavior caused by closing a tab recently made parent by having its parent tab closed?

### DISCUSSION THREAD

**Discussion by ErroneousDictation** [piroor/treestyletab#3042] [Q&A] [answered]
https://gfycat.com/ThatNearAmericantoad
Gif of the issue I'm having. Essentially if I immediately close a tab that was just promoted a level by having its parent closed, all of the children of the promoted tabs are also closed. 
https://gfycat.com/IdolizedUncommonBarbet
Gif showing that the behavior does NOT occur if I close the two tabs slowly. In that case, the children tabs are preserved and this is my PREFERRED functionality.

In both cases I am only clicking the close button twice. The third tab closing seems to depend on how fast the parent tabs are closed. Is there an option somewhere to make tabs always behave like in the latter scenario, or is the former scenario a bug that needs to be reported?

**Comment by irvinm** [comment]
I see the same behavior as you do.  I would write this up as a defect.  There "might" be some type of timing value somewhere in the "All configs" section, but this might just be a defect as well.

**Comment by piroor** [comment, marked as answer]
Basically it is not an expected behavior, and TST does not provide any option about that. On an expanded tree, closing of a parent tab with its closebox must close only the individual tab instead of the whole tree.

Hmm, I couldn't reproduce the problem but I guess that TST may misdetect the secondary closed tab as a collapsed parent unexpectedly.

I've opened an issue #3045 based on this topic. After it become fixed this discussion should be deleted.

**Reply by piroor** [reply]
This problem was fixed at TST 3.8.17 with the commit 8b713b5. Please see #3045 for more details.

**Comment by irvinm** [comment]
@piroor I think you can mark this as solved.

**Reply by piroor** [reply]
Thanks! I didn't know that GitHub Discussions does not provide ability to "close" a discussion. Anyway I've marked my reply as the answer of this topic.
